### PR TITLE
fix tree model description

### LINKED ISSUE #494: Missing info from model_description() for regression trees
## Versions

**river version**: master 03.03.2021
**Python version**: 3.8.8
**Operating system**: Windows 10 Pro v2004

## Describe the bug

When using
```python
tree_regressor.model_description()
``` 

This is the actual output:
```python
if you_gov <= -0.2422019367906697:
  if morning_consult <= -0.5832368018625002:
    Leaf = Output {Mean | Var}
  if morning_consult > -0.5832368018625002:
    Leaf = Output {Mean | Var}
if you_gov > -0.2422019367906697:
  if ipsos <= 0.02250866367988844:
    Leaf = Output {Mean | Var}
  if ipsos > 0.02250866367988844:
    if morning_consult <= -0.9803256024230158:
      Leaf = Output {Mean | Var}
    if morning_consult > -0.9803256024230158:
      if gallup <= 0.4793608626664672:
        Leaf = Output {Mean | Var}
      if gallup > 0.4793608626664672:
        Leaf = Output {Mean | Var}
```

And this, I suppose, is the expected output:

```python
if you_gov <= -0.2422019367906697:
  if morning_consult <= -0.5832368018625002:
    Leaf = Output {38.742721186666664 | 1.8713735070001474}
  if morning_consult > -0.5832368018625002:
    Leaf = Output {39.01400445714287 | 1.7417825868271475}
if you_gov > -0.2422019367906697:
  if ipsos <= 0.02250866367988844:
    Leaf = Output {40.99789951572325 | 1.8287692797915924}
  if ipsos > 0.02250866367988844:
    if morning_consult <= -0.9803256024230158:
      Leaf = Output {41.76915201960785 | 1.1050195230278164}
    if morning_consult > -0.9803256024230158:
      if gallup <= 0.4793608626664672:
        Leaf = Output {41.95805093478262 | 0.12889200096775508}
      if gallup > 0.4793608626664672:
        Leaf = Output {41.78473850719421 | 0.8817032902779682}
```

Since for classifier models the following holds:

Input:
```python
from river import datasets
from river import tree

model = tree.HoeffdingTreeClassifier(
    grace_period=5, split_confidence=1e-5, split_criterion="gini", max_depth=10, tie_threshold=0.05,
)
for x, y in datasets.Phishing():
    model = model.learn_one(x, y)

print(model.model_description())
```
Output:
```python
if empty_server_form_handler <= 0.5454545454545454:
  Leaf = Class True | {True: 448.4948683707587, False: 58.946812947101755}
if empty_server_form_handler > 0.5454545454545454:
  if https <= 0.5454545454545454:
    Leaf = Class False | {False: 69.36506478686493, True: 39.0}
  if https > 0.5454545454545454:
    Leaf = Class False | {False: 307.63493521313507, True: 10.0}
```

------
I also suppose the same behaviour (property name instead of its value) occures in detrenders/metrics/other tree models/other representations (e.g. representation for Multi-Target regression, just a few lines above or `.debug_one()` method), but I cannot quite put my finger on them for now.

## Steps/code to reproduce
```python
from river import datasets
from river import evaluate
from river import metrics
from river import tree
from river import preprocessing

dataset = datasets.TrumpApproval()
tree_regressor = tree.HoeffdingAdaptiveTreeRegressor(
    grace_period=50, leaf_prediction="adaptive", model_selector_decay=0.3, seed=0
)

model = preprocessing.StandardScaler() | tree_regressor

metric = metrics.MAE()

evaluate.progressive_val_score(dataset, model, metric)

print(tree_regressor.model_description())
```

### PR DESCRIPTION
**Pull request by ColdTeapot273K**
fixes #494

### CODE REVIEW

File: river/tree/_nodes/base.py [hunk 1]
```diff
@@ -145,7 +145,7 @@ def describe_subtree(self, tree, buffer: List[str], indent: int = 0):
                     text += f"{target_id}: {self.stats[target_id].mean} | {self.stats[target_id]}"
                     text += ", " if i < len(self.stats) - 1 else ""
             else:  # Single-target regression
-                text += f"{self.stats.mean} | {self.stats}"
+                text += f"{self.stats.mean.get()} | {self.stats.get()}"
```
Comment (MaxHalford) [inline river/tree/_nodes/base.py:148]: Could you instead do `{repr(self.stats.mean)} | {repr(self.stats)`. This was we will also get the `Mean` and `Var` labels.

Comment (smastelini) [reply river/tree/_nodes/base.py:148]: Good catch!

Comment (ColdTeapot273K) [reply river/tree/_nodes/base.py:148]: I tried {repr(self.stats.mean)} | {repr(self.stats)} ` and got this:
```
Leaf = Output {Mean () | Var (
  ddof=1
)}
```

{repr(self.stats.mean.get())} | {repr(self.stats.get())} `  gives values only. 

I've hardcoded the labels like so `f"Mean={self.stats.mean.get()} | Var={self.stats.get()}"` 
to have output like this:
`Leaf = Output {Mean=0.0 | Var=0.0}`

as a short-term solution. 

Thoughts?

Comment (smastelini) [reply river/tree/_nodes/base.py:148]: I can confirm that the repr call is not working as we would expect.

Comment (smastelini) [reply river/tree/_nodes/base.py:148]: Here's our culprit:

https://github.com/online-ml/river/blob/de68efe1afed3f95344674e613abad5146a9541c/river/stats/base.py#L20

Univariate should inherit from `Statistic`. Do you confirm this, @MaxHalford?

Comment (MaxHalford) [reply river/tree/_nodes/base.py:148]: Good catch!

Comment (smastelini) [reply river/tree/_nodes/base.py:148]: I'll submit a fix. Once it is in place, I'll ask you, @ColdTeapot273K to pull the changes.

Comment (smastelini) [reply river/tree/_nodes/base.py:148]: @ColdTeapot273K, please pull the latest changes from `master`, then you can use Max's suggestion and everything should work fine.

Thanks once again for checking that!

**Comment by smastelini** [comment]
Hi @ColdTeapot273K, there's only one missing piece. We have some automated solutions to ensure the standardization of our codebase. They are the cause of your failing tests.

Please follow the instructions in our [contribution guidelines](https://github.com/online-ml/river/blob/master/CONTRIBUTING.md#local-setup). You need to install and run the `pre-commit actions` to format the code. But before doing that, make sure that [pre-commit](https://pre-commit.com/) is installed in your machine.

**Comment by ColdTeapot273K** [comment]
@smastelini yeah, of course. I just have custom `black` config and forgot to switch for this project, which gave some trouble, And now it's new line check failed. Worst case I'll just run the failed hooks. Probably even today.

IIRC I had problem with some hooks related to testing before so uninstalled them completely to make commits (tests failed literaly on pristine checkout from upstream). I shall try them again and report if they give errors. Might be OS related since I'm currently on Win10

**Comment by smastelini** [comment]
At some time, if I'm not wrong, I had to install the development version of `pre-commit` to solve some issues. Maybe this could be the case.

**Comment by ColdTeapot273K** [comment]
@smastelini pre-push hook is causing the trouble:
```log
> git push origin fix/stats-representation:fix/stats-representation
> git ls-tree -l HEAD -- E:\MoreFun\develop-river\river\tree\hoeffding_tree.py
/usr/bin/env: ‘python’: Permission denied
error: failed to push some refs to 'https://github.com/ColdTeapot273K/river.git'
```

Had to delete that hook speceficially to make a push

(Using Git Bash shell on Win10 and Conda. Can share my .bashrc and .bash_profile if needed. Not so fluent in shell scripting to troubleshoot this easily.)

**Comment by smastelini** [comment]
Thanks, @ColdTeapot273K for looking into that. This PR is ready to be merged.

**Pull request merged**